Documentation/src/case-studies/calendar-code.md

```diff
@@ -0,0 +1,110 @@
+﻿
+## Program.cs
+
+The complete calendar program:
+
+```cs
+using System.Collections.Immutable;
+using Funcky.Extensions;
+using System.Globalization;
+using Funcky;
+using Funcky.Monads;
+using static Funcky.Functional;
+
+namespace Calendar;
+
+internal static class StringExtensions
+{
+    public static string Center(this string text, int width)
+        => (width - text.Length) switch
+        {
+            0 => text,
+            1 => $" {text}",
+            _ => Center($" {text} ", width),
+        };
+}
+
+internal class Program
+{
+    private const int FirstDayOfTheWeek = 0;
+    private const int HorizontalMonths = 3;
+    private const int DaysInAWeek = 7;
+    private const int WidthOfDay = 3;
+    private const int WidthOfAWeek = WidthOfDay * DaysInAWeek;
+    private const string MonthNameFormat = "MMMM";
+
+    private static void Main(string[] args)
+        => Console.WriteLine(CreateCalendarString(ExtractYear(args).GetOrElse(DateTime.Now.Year)));
+
+    private static Option<int> ExtractYear(IEnumerable<string> args)
+        => args.FirstOrNone()
+            .AndThen(ParseExtensions.ParseInt32OrNone);
```

Comment:
Why not select?

Reply:
AndThen has convenient overloads, this actually is a SelectMany not a Select.